### PR TITLE
build(attributable-map): convert to esm

### PR DESCRIPTION
**Pull request by DLehenbauer**
Mechanical conversion to esm

### CODE REVIEW

File: examples/apps/attributable-map/package.json [hunk 1]
```diff
@@ -49,21 +47,15 @@
...
 		"html-webpack-plugin": "^5.5.0",
-		"jest": "^29.6.2",
-		"jest-environment-puppeteer": "^9.0.2",
-		"jest-junit": "^10.0.0",
-		"jest-puppeteer": "^9.0.2",
 		"prettier": "~3.0.3",
 		"process": "^0.11.10",
 		"puppeteer": "^22.2.0",
 		"rimraf": "^4.4.0",
-		"ts-jest": "^29.1.1",
```
Comment (DLehenbauer) [inline examples/apps/attributable-map/package.json:66]: Example has no tests.  Jest dependencies and config are unused.

File: examples/apps/attributable-map/package.json [hunk 2]
```diff
@@ -49,21 +47,16 @@
 		"@fluidframework/build-tools": "^0.34.0",
 		"@fluidframework/eslint-config-fluid": "^4.0.0",
 		"@fluidframework/test-tools": "^1.0.195075",
-		"@types/jest": "29.5.3",
+		"@types/jest-environment-puppeteer": "workspace:",
```
Comment (jason-ha) [inline examples/apps/attributable-map/package.json:50]: delete instead?
@types/node dependency is also suspect without any testing.
And just below that cross-env appears unused. And there are two more puppeteer deps remain.

File: examples/apps/attributable-map/tsconfig.json [hunk 3]
```diff
@@ -1,9 +1,7 @@
 {
-	"extends": "@fluidframework/build-common/ts-common-config.json",
+	"extends": "../../../common/build/build-common/tsconfig.node16.json",
 	"compilerOptions": {
-		"module": "esnext",
-		"outDir": "lib",
-		"jsx": "react",
+		"outDir": "./lib",
 		"types": ["jest", "puppeteer", "jest-environment-puppeteer", "expect-puppeteer"],
```
Comment (jason-ha) [inline examples/apps/attributable-map/tsconfig.json:5]: types look non-viable with deps removed

File: examples/apps/attributable-map/webpack.config.cjs [hunk 4]
```diff
@@ -32,7 +32,7 @@ module.exports = (env) => {
 			},
 			output: {
 				filename: "[name].bundle.js",
-				path: path.resolve(__dirname, "dist"),
+				path: path.resolve(__dirname, "lib"),
```
Comment (jason-ha) [inline examples/apps/attributable-map/webpack.config.cjs:35]: We should not mix tsc output and webpack output. Prefer a rename to have a webpack directory, but okay to leave as dist for a later general webpack config cleanup.
You can test this out by webpacking (just build or just serve(start)) without doing any tsc.

**Review by jason-ha — CHANGES_REQUESTED** [review]

**Review by jason-ha — APPROVED** [review]

**Pull request merged**